src/ape/managers/compilers.py

```diff
@@ -99,16 +99,27 @@ def compile(self, contract_filepaths: List[Path]) -> Dict[str, ContractType]:
                 paths_to_compile, base_path=self.config_manager.contracts_folder
             )
             for contract_type in compiled_contracts:
-
-                if contract_type.name in contract_types_dict:
-                    raise CompilerError(
-                        "ContractType collision across compiler plugins "
-                        f"with contract: {contract_type.source_id}"
+                contract_name = contract_type.name
+                if not contract_name:
+                    if contract_type.source_id:
+                        logger.warning(f"Contract '{contract_type.source_id}' missing name.")
+                    else:
+                        logger.warning(f"Empty contract type found in compiler '{extension}'.")
```

Comment:
We had some issues in `ape-pm` but I fixed them there, so these warnings don't happen anymore. **I agree these should be error states!** So I changed them to raise errors here.

This would still happen for custom compiler plugins that are poorly implemented, so I adjusted the warning to describe that:

```python
  raise CompilerError(
      f"Contract '{contract_type.source_id}' missing name. "
      f"Was compiler plugin for '{extension} implemented correctly?"
  )
```